File: Data Structures & Algorithms/task-scheduling/submission-2.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] freq = new int[26];
        int max_freq=0;
        for(char t:tasks){
            freq[t-'A']++;
            max_freq=Math.max(max_freq,freq[t-'A']);
        }
        int max_count=-1;
        for(int f:freq)
            if(f==max_freq) max_count++;

        int ans = (max_freq-1)*n+max_freq+max_count;

        return Math.max(tasks.length,ans);

    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-2.cpp
class Solution {
    int fct(vector<int>& nums , int l , int r){
        int n = r-l+1;
                if (n == 0) return 0;
                if (n == 1) return nums[l];

                vector<int> dp(n, 0);
                dp[0] = nums[l];
                dp[1] = max(nums[l], nums[l+1]);

                for (int i = 2; i < n; i++) {
                    dp[i] = max(dp[i - 1], dp[i - 2] + nums[l+i]);
                }

                return dp[n - 1];
    }
public:
    int rob(vector<int>& nums) {
        int n=nums.size();
        if (n == 0) return 0;
        if (n == 1) return nums[0];
        return max(fct(nums ,0,n-2),fct(nums ,1,n-1));
        
    }
};